python_modules/libraries/dagster-aws/dagster_aws/ecs/executor.py

```diff
@@ -0,0 +1,333 @@
+    """Executor which launches steps as ECS tasks."""
+    run_launcher = init_context.instance.run_launcher
+
+    check.invariant(
+        isinstance(run_launcher, EcsRunLauncher),
+        "Using the ecs_executor currently requires that the run be launched in an ECS task via the EcsRunLauncher.",
+    )
+
+    exc_cfg = init_context.executor_config
+
+    return StepDelegatingExecutor(
+        EcsStepHandler(
+            run_task_kwargs=exc_cfg.get("run_task_kwargs"),  # type: ignore
+            cpu=exc_cfg.get("cpu"),  # type: ignore
+            memory=exc_cfg.get("memory"),  # type: ignore
+            ephemeral_storage=exc_cfg.get("ephemeral_storage"),  # type: ignore
+            task_overrides=exc_cfg.get("task_overrides"),  # type:ignore
+        ),
+        retries=RetryMode.from_config(exc_cfg["retries"]),  # type: ignore
+        max_concurrent=check.opt_int_elem(exc_cfg, "max_concurrent"),
+        tag_concurrency_limits=check.opt_list_elem(exc_cfg, "tag_concurrency_limits"),
+        should_verify_step=True,
+    )
+
+
+class EcsStepHandler(StepHandler):
+    @property
+    def name(self):
+        return "EcsStepHandler"
+
+    def __init__(
+        self,
+        run_task_kwargs: Optional[Mapping[str, Any]],
+        cpu: Optional[int],
+        memory: Optional[int],
+        ephemeral_storage: Optional[int],
+        task_overrides: Optional[Mapping[str, Any]],
+    ):
+        super().__init__()
+
+        run_task_kwargs = run_task_kwargs or {}
+
+        self.ecs = boto3.client("ecs")
+        self.ec2 = boto3.resource("ec2")
+
+        # confusingly, run_task expects cpu and memory value as strings
+        self._cpu = str(cpu) if cpu else None
+        self._memory = str(memory) if memory else None
+
+        self._ephemeral_storage = ephemeral_storage
+        self._task_overrides = check.opt_mapping_param(task_overrides, "task_overrides")
+
+        current_task_metadata = get_current_ecs_task_metadata()
+        current_task = get_current_ecs_task(
+            self.ecs, current_task_metadata.task_arn, current_task_metadata.cluster
+        )
+        task_kwargs = get_task_kwargs_from_current_task(
+            self.ec2,
+            current_task_metadata.cluster,
+            current_task,
+        )
+        self._cluster_arn = current_task["clusterArn"]
+        self._task_definition_arn = current_task["taskDefinitionArn"]
+        self._run_task_kwargs = {
+            **task_kwargs,
+            **run_task_kwargs,
+            "taskDefinition": current_task["taskDefinitionArn"],
+        }
+
+    def _get_run_task_kwargs(
+        self,
+        run: DagsterRun,
+        args: Sequence[str],
+        step_key: str,
+        step_tags: Mapping[str, str],
+        step_handler_context: StepHandlerContext,
+        container_context: EcsContainerContext,
+    ):
+        run_launcher = cast(EcsRunLauncher, step_handler_context.instance.run_launcher)
```

Comment:
lets use more things from the run launcher here when possible, see above - if you have specified what cluster to use in your dagster.yaml for the run launcher we should be able to pull that here has well rather than re-deriving it from the current task